### PR TITLE
docs(showcase): Add reflex to showcase

### PR DESCRIPTION
**Pull request by tgberkeley**
## What is the purpose of this pull request?

- [ ] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [x] Other:

### Description
This PR adds the company Reflex (https://reflex.dev) to the Showcase page. This Repo has over 20,000 GitHub stars and uses Lucide icons as its only icon type in the library.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by tgberkeley** [comment]
@jguddas what is the reason for the 👎 above, am I missing something in this PR?

File: docs/public/company-logos/reflex-dark.svg
Comment (karsa-mistmere) [inline docs/public/company-logos/reflex-dark.svg:1]: You've mixed up light with dark:
image
image

And maybe "Used in" would be the more appropriate category.

I'll leave the decision up to @ericfennis

Comment (tgberkeley) [reply docs/public/company-logos/reflex-dark.svg:1]: @karsa-mistmere yes I noticed this and fixed it in a follow up commit

Comment (karsa-mistmere) [reply docs/public/company-logos/reflex-dark.svg:1]: > @tgberkeley Can you confirm, lucide is part of the tool or is it only used on the docs page?

I think part of the tool: https://reflex.dev/docs/library/data-display/icon/

**Comment by jguddas** [comment]
> @jguddas what is the reason for the 👎 above, am I missing something in this PR?

They have fewer users than us, so it's more of an advertisement for them than it is for us.

I would rather expect there to be companies such as [adobe](https://github.com/search?q=org%3Aadobe%20lucide&type=code).

But all in all I'm very much against having such a section, it's legally and morally a pretty dark gray area.

File: docs/public/company-logos/reflex-dark.svg
Comment (ericfennis) [reply docs/public/company-logos/reflex-dark.svg:1]: In that case this should be moved to the "used in" section.

**Review by ericfennis — CHANGES_REQUESTED** [review]
Should be moved to the "Used in" section.

**Comment by tgberkeley** [comment]
@ericfennis just moved it to the used in section

**Comment by tgberkeley** [comment]
@ericfennis is there anything else here that I need to edit to get this PR approved?

**Review by ericfennis — APPROVED** [review]
LGTM!

**Pull request merged**